pyvista/trame/ui.py

```diff
@@ -0,0 +1,337 @@
+                    )
+                    button(
+                        click=viewer.view_xy,
+                        icon='mdi-axis-z-arrow',
+                        tooltip='Reset Camera Z',
+                    )
+                    vuetify.VDivider(vertical=True, classes='mx-1')
+                    checkbox(
+                        model=(viewer.OUTLINE, False),
+                        icons=('mdi-cube', 'mdi-cube-off'),
+                        tooltip=f"Toggle bounding box ({{{{ {viewer.OUTLINE} ? 'on' : 'off' }}}})",
+                    )
+                    checkbox(
+                        model=(viewer.GRID, False),
+                        icons=('mdi-ruler-square', 'mdi-ruler-square'),
+                        tooltip=f"Toggle ruler ({{{{ {viewer.GRID} ? 'on' : 'off' }}}})",
+                    )
+                    # Server rendering options
+                    if mode == 'trame':
+                        vuetify.VDivider(vertical=True, classes='mx-1')
+                        checkbox(
+                            model=(viewer.SERVER_RENDERING, default_server_rendering),
+                            icons=('mdi-dns', 'mdi-open-in-app'),
+                            tooltip=f"Toggle rendering mode ({{{{ {viewer.SERVER_RENDERING} ? 'remote' : 'local' }}}})",
+                        )
+                    with vuetify.VRow(
+                        v_show=(viewer.SERVER_RENDERING, default_server_rendering),
+                        classes='pa-0 ma-0 align-center',
+                    ):
+                        checkbox(
+                            model=(viewer.AXIS, False),
+                            icons=('mdi-axis-arrow-info', 'mdi-axis-arrow-info'),
+                            tooltip=f"Toggle axis ({{{{ {viewer.AXIS} ? 'on' : 'off' }}}})",
+                        )
+                        button(
+                            # Must use single-quote string for JS here
+                            click=f"utils.download('screenshot.png', trigger('{viewer.SCREENSHOT}'), 'image/png')",
+                            icon='mdi-file-png-box',
+                            tooltip='Save screenshot',
+                        )
+        if mode == 'trame':
+            view = PyVistaRemoteLocalView(
+                plotter,
+                mode=(
+                    # Must use single-quote string for JS here
+                    f"{viewer.SERVER_RENDERING} ? 'remote' : 'local'",
+                    'remote' if default_server_rendering else 'local',
+                ),
+                **kwargs,
+            )
+            ctrl.view_update_image = view.update_image
+        elif mode == 'server':
+            view = PyVistaRemoteView(plotter, **kwargs)
+        elif mode == 'client':
+            view = PyVistaLocalView(plotter, **kwargs)
+        ctrl.view_update = view.update
+        ctrl.view_reset_camera = view.reset_camera
+        ctrl.view_push_camera = view.push_camera
```

Comment:
the last instance of the viewer will override the other ctrl. methods...

Reply:
Oh woah. I'm surprised I didn't notice this...

Reply:
we can chat next week so I can tell you how to better handle it...